src/components/Modal.js

```diff
@@ -120,30 +122,46 @@ export default class Modal extends Component {
     !isReact16 && this.renderPortal(this.props);
   }
 
-  componentWillReceiveProps(newProps) {
-    if (!canUseDOM) return;
+  static getDerivedStateFromProps(newProps, prevState) {
+    if (!canUseDOM) return null;
     const { isOpen } = newProps;
     // Stop unnecessary renders if modal is remaining closed
-    if (!this.props.isOpen && !isOpen) return;
+    if (prevState && !prevState.isOpen && !isOpen) return null;
 
-    const currentParent = getParentElement(this.props.parentSelector);
+    const currentParent = getParentElement(prevState.parentSelector);
     const newParent = getParentElement(newProps.parentSelector);
 
+    const enumeratedState = {
+      isOpen: newProps.isOpen,
+      parentSelector: newProps.parentSelector,
+      portalClassName: newProps.portalClassName
+    };
+
     if (newParent !== currentParent) {
       currentParent.removeChild(this.node);
       newParent.appendChild(this.node);
     }
 
+    const newState = Object.assign(prevState, enumeratedState);
     !isReact16 && this.renderPortal(newProps);
+    return newState;
   }
 
-  componentWillUpdate(newProps) {
-    if (!canUseDOM) return;
-    if (newProps.portalClassName !== this.props.portalClassName) {
-      this.node.className = newProps.portalClassName;
+  getSnapshotBeforeUpdate(prevProps, prevState) {
+    if (!canUseDOM) {
+      return null;
+    } else {
+      this.node.className = prevState.portalClassName;
```

Comment:
this method should be used for reading DOM not for mutating it

Reply:
See https://reactjs.org/docs/react-component.html#getsnapshotbeforeupdate and https://twitter.com/dan_abramov/status/981712092611989509

Reply:
@Hypnosphi I did the exact same thing you mentioned before making this PR, and yes I read the links already. But the problem I am facing is that cDU is not at all called(not with the `snapshot` parameter). Also it is not documented as to how to use the snapshot in react properly. This could most probably be a bug in `pollyfil-react-compat` library or I am doing something wrong. Please feel free to make a PR on this.

Reply:
There's no usecase for snapshotting here, because you don't read any data from DOM, you just change a className. It should work OK in cDU:

```
componentDidUpdate(prevProps) {
    if (!canUseDOM) return;
    if (prevProps.portalClassName !== this.props.portalClassName) {
      this.node.className = this.props.portalClassName;
    }
  }
```

Reply:
@Hypnosphi sorry I didn't get the time to get back to this. Yes this makes sense as done in #642 .